tracing-mock/src/span.rs

```diff
@@ -760,3 +886,72 @@ impl fmt::Debug for NewSpan {
         s.finish()
     }
 }
+
+impl PartialEq for ExpectedId {
+    fn eq(&self, other: &Self) -> bool {
+        let own_id = self.inner.load(Ordering::Relaxed);
+        let other_id = other.inner.load(Ordering::Relaxed);
+
+        own_id != Self::UNSET && own_id == other_id
+    }
+}
+
+impl Eq for ExpectedId {}
+
+impl ExpectedId {
+    const UNSET: u64 = 0;
+
+    pub(crate) fn new_unset() -> Self {
+        Self {
+            inner: Arc::new(AtomicU64::from(Self::UNSET)),
+        }
+    }
+
+    pub(crate) fn set(&self, span_id: u64) -> Result<(), SetActualSpanIdError> {
+        self.inner
+            .compare_exchange(Self::UNSET, span_id, Ordering::Relaxed, Ordering::Relaxed)
+            .map_err(|current| SetActualSpanIdError {
+                previous_span_id: current,
+                new_span_id: span_id,
+            })?;
+        Ok(())
+    }
+
+    pub(crate) fn check(&self, actual: u64, ctx: fmt::Arguments<'_>, collector_name: &str) {
+        let id = self.inner.load(Ordering::Relaxed);
+
+        assert!(
+            id != Self::UNSET,
+            "\n[{}] expected {} to have expected ID set, but it hasn't been, \
+            perhaps this `ExpectedId` wasn't used in a call to `MockCollector::new_span()`?",
+            collector_name,
+            ctx,
+        );
+
+        assert_eq!(
+            id, actual,
+            "\n[{}] expected {} to have ID `{}`, but it has `{}` instead",
+            collector_name, ctx, id, actual,
+        );
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct SetActualSpanIdError {
+    previous_span_id: u64,
+    new_span_id: u64,
+}
+
+impl fmt::Display for SetActualSpanIdError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "Could not set `ExpecedId` to {new}, \
+               it had already been set to {previous}",
```

Comment:
Nit, there seems to be something weird going on with the leading whitespace

Reply:
Fixed.